homeassistant/components/aussie_broadband/config_flow.py

```diff
@@ -0,0 +1,184 @@
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.data: dict = {}
+        self.options: dict = {CONF_SERVICES: []}
+        self.services: list[dict[str]] = []
+        self.client: AussieBB | None = None
+        self._reauth_username: str | None = None
+
+    async def async_auth(self, user_input: dict[str, str]) -> dict[str, str] | None:
+        """Reusable Auth Helper."""
+        self.client = AussieBB(
+            user_input[CONF_USERNAME],
+            user_input[CONF_PASSWORD],
+            async_get_clientsession(self.hass),
+        )
+        try:
+            await self.client.login()
+            return None
+        except AuthenticationException:
+            return {"base": "invalid_auth"}
+        except ClientError:
+            return {"base": "cannot_connect"}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] | None = None
+        if user_input is not None:
+            if not (errors := await self.async_auth(user_input)):
+                await self.async_set_unique_id(user_input[CONF_USERNAME].lower())
+                self._abort_if_unique_id_configured()
+
+                self.data = user_input
+                self.services = await self.client.get_services()  # type: ignore[union-attr]
+
+                if not self.services:
+                    return self.async_abort(reason="no_services_found")
+
+                if len(self.services) == 1:
+                    return self.async_create_entry(
+                        title=self.data[CONF_USERNAME],
+                        data=self.data,
+                        options={CONF_SERVICES: [str(self.services[0][SERVICE_ID])]},
+                    )
+
+                # Account has more than one service, select service to add
+                return await self.async_step_service()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_service(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the optional service selection step."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.data[CONF_USERNAME], data=self.data, options=user_input
+            )
+
+        service_options = {str(s[SERVICE_ID]): s["description"] for s in self.services}
+        return self.async_show_form(
+            step_id="service",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SERVICES, default=list(service_options.keys())
```

Comment:
Why do we need the user to select the services? Why not create sensors for all available services? The user can disable the entities they aren't interested in.

Reply:
This was discussed https://github.com/home-assistant/core/pull/53552#discussion_r699018653

As per https://github.com/home-assistant/architecture/blob/master/adr/0003-monitor-condition-and-data-selectors.md
each service requires a unique API calls, so if we enabled every service by default but a user only wants 1 service, then the integration would be making extra API calls on every refresh.

Reply:
Since we create one coordinator per service and the coordinator only does automatic refresh as long as there are listeners (enabled entities), I don't think that's a good reason.

I suggest we create entities for all services. Some of the entities can be disabled by default if we think we know to do that selection.